src/SpectatorView.Native/SpectatorView.Compositor/Compositor/DeckLinkDevice.cpp

```diff
@@ -463,11 +463,13 @@ bool DeckLinkDevice::Init(ID3D11ShaderResourceView* colorSRV, ID3D11Texture2D* o
     if (colorSRV != nullptr)
     {
         colorSRV->GetDevice(&device);
-    }
 
-    // Get input interface
-    if (m_deckLink->QueryInterface(IID_IDeckLinkInput, (void**)&m_deckLinkInput) != S_OK)
-        return false;
+        // Get input interface
+        if (m_deckLink->QueryInterface(IID_IDeckLinkInput, (void**)&m_deckLinkInput) != S_OK)
+            return false;
+    }
+    if (device == nullptr && outputTexture != nullptr)
+        outputTexture->GetDevice(&device);
```

Comment:
Can you add a comment on why this is necessary (e.g. when it will happen that either colorSRV is null or colorSRV fails to get a device)? It wasn't obvious to me reading the code.

Reply:
Ah, I think I understand, this happens for the output-only frame, because it doesn't have input.  In that case, I think it would be more clear for this second if to be an "else if"

---
In reply to: [404388384](https://github.com/microsoft/MixedReality-SpectatorView/pull/357#discussion_r404388384) [](ancestors = 404388384)